Add ONIE partition types

Copyright (C) 2014 Curt Brune <curt@cumulusnetworks.com>

SPDX-License-Identifier:     GPL-2.0

Add GUID for two ONIE partition types:

   // Open Network Install Environment (ONIE) Boot Partition
   AddType(0x3100, "7412F7D5-A156-4B13-81DC-867174929325", "ONIE boot");

   // Open Network Install Environment (ONIE) Configuration Partition
   AddType(0x3100, "D4E6E2CD-4469-46F3-B5CB-1BFF57AFC149", "ONIE config");

diff --git a/parttypes.cc b/parttypes.cc
index 87f5c86..f78489b 100644
--- a/parttypes.cc
+++ b/parttypes.cc
@@ -91,6 +91,11 @@ void PartType::AddAllTypes(void) {
    AddType(0x1e00, "EBD0A0A2-B9E5-4433-87C0-68B6B72699C7", "Microsoft basic data", 0); // Hidden FAT-16 LBA
    AddType(0x2700, "DE94BBA4-06D1-4D40-A16A-BFD50179D6AC", "Windows RE");
 
+   // Open Network Install Environment (ONIE) specific types.
+   // See http://www.onie.org/
+   AddType(0x3000, "7412F7D5-A156-4B13-81DC-867174929325", "ONIE boot");
+   AddType(0x3001, "D4E6E2CD-4469-46F3-B5CB-1BFF57AFC149", "ONIE config");
+
    // PowerPC reference platform boot partition
    AddType(0x4100, "9E1A2D38-C612-4316-AA26-8B49521E5A8B", "PowerPC PReP boot");
 
